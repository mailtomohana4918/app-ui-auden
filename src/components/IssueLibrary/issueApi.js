import axios from "axios";
export const issueApi = {
  getIssues: async (name, owner) => {
    try {
      const response = await axios.get(
        `https://api.github.com/repos/${owner}/${name}/issues`
      );
      //alert(JSON.stringify(response.data))
      return response.data;
    } catch (error) {
      console.log(error);
      throw error;
    }
  },
};
