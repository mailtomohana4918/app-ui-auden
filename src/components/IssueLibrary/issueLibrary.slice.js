import { createSlice } from "@reduxjs/toolkit";

const initialState = {
  issueData: [],
};

export const issueSlice = createSlice({
  name: "issue",
  initialState,
  reducers: {
    pushIssuestoStore: (state, action) => {
      state.issueData.push(action.payload);
    },
  },
});

export const { pushIssuestoStore } = issueSlice.actions;
