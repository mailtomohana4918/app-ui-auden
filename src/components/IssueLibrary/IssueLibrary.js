import React, { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { pushIssuestoStore } from "./issueLibrary.slice";
import { issueApi } from "./issueApi";
import CustomTable from "../shared/CustomTable/CustomTable";
import Header from "../shared/Header/Header";
import { Grid, Box, Button } from "@material-ui/core";
const columns = [
  { id: "title", label: "Issue", minWidth: 100 },
  { id: "number", label: "Issue Number", minWidth: 100 },
  {
    id: "state",
    label: "Status",
    minWidth: 170,
    align: "right",
    format: (value) => value.toLocaleString("en-US"),
  },
  {
    id: "created_at",
    label: "Created Date",
    minWidth: 170,
    align: "right",
    format: (value) => value,
  },
  {
    id: "updated_at",
    label: "Last Updated Date",
    minWidth: 170,
    align: "right",
    format: (value) => value,
  },
];

const IssueLibrary = (props) => {
  const stateData = useSelector((state) => state.search.searchData);
  const dispatch = useDispatch();
  const [dataRecieved, setDataReceived] = useState(false);
  const [filteredDataIssue, setFiltertedDataIssue] = useState();
  const issueDetails = useSelector((state) => state.issue.issueData);
  const fetchIssues = async (name, owner) => {
    const response = await issueApi.getIssues(name, owner);
    const id = Number(props.match.params.id);
    const updatedResponse = { id: id, issueDetails: [...response] };
    dispatch(pushIssuestoStore(updatedResponse));
    setDataReceived(true);
  };

  useEffect(
    () => {
      const repoData = stateData.find(
        (el) => Number(el.id) === Number(props.match.params.id)
      );

      const filteredData = fetchFromStore(Number(props.match.params.id));
      if (filteredData === undefined) {
        fetchIssues(repoData.name, repoData.owner.login);
      } else {
        setDataReceived(true);
        setFiltertedDataIssue(filteredData);
      }
    },
    []
  );

  const fetchFromStore = (id) => {

    const filteredData = issueDetails.find((el) => el.id === id);

    return filteredData;
  };

  const IssueContainer = () => {
    return (
      
      <CustomTable
        searchResult={
          filteredDataIssue === undefined
            ? issueDetails.find((el) => el.id === Number(props.match.params.id))
                .issueDetails
            : filteredDataIssue.issueDetails
        }
        columns={columns}
      />
    );
  };

  return (
    <>
      <Header />

      {dataRecieved && (
        <>
          <Grid
            container
            spacing={0}
            direction="column"
            alignItems="center"
            justify="center"
          >
            <Box paddingTop="120px">
              <IssueContainer />
            </Box>
            <Box paddingTop="20px">
              <Button
                variant="contained"
                color="primary"
                onClick={() => props.history.push("/")}
              >
                Close
              </Button>
            </Box>
          </Grid>
        </>
      )}
    </>
  );
};

export default IssueLibrary;
