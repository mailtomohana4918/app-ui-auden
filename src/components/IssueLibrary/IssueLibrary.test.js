import { render } from "@testing-library/react";
import IssueLibrary from "./IssueLibrary";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
describe("IssueLibrary", () => {
  const issue_Date = "2020-07-27T16:48:02Z";
  const initialState = {
    issue: {
      issueData: [
        {
          id: 1,
          issueDetails: [
            {
              id: 1,
              title: "Component re-render",
              state: 2000,
              created_at: issue_Date,
              updated_at: issue_Date,
            },

            {
              id: 2,
              title: "Chaining Issue",
              state: 2000,
              created_at: issue_Date,
              updated_at: issue_Date,
            },
            {
              id: 3,
              title: "Media query not rendering",
              state: 2000,
              created_at: issue_Date,
              updated_at: issue_Date,
            },
            {
              id: 4,
              title: "snapshots not updating",
              state: 2000,
              created_at: issue_Date,
              updated_at: issue_Date,
            },
          ],
        },
      ],
    },
    search: {
      searchData: [
        { id: 1, name: "React", watchers: 2000, forks: 100, open_issues: 200 },

        { id: 2, name: "Vue", watchers: 3000, forks: 200, open_issues: 300 },
        {
          id: 3,
          name: "Angular",
          watchers: 4000,
          forks: 300,
          open_issues: 400,
        },
        { id: 4, name: "Django", watchers: 5000, forks: 400, open_issues: 500 },
      ],
    },
  };
  const mockStore = configureStore();
  let store, wrapper;

  it("render component as expected", () => {
    store = mockStore(initialState);
    const { asFragment } = render(
      <Provider store={store}>
        <Router
          render={(routeProps) => <IssueLibrary {...routeProps} id={1} />}
        />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
