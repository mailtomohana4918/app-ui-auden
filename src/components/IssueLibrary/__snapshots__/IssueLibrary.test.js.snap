// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueLibrary render component as expected 1`] = `<DocumentFragment />`;
